### PR TITLE
Move to a remote approach for environment templates

### PR DESCRIPTION
**Pull request by timfpark**
Closes https://github.com/Microsoft/bedrock/issues/268.  Moves gitops provisioning related logic to `azure/azure-gitops` and migrates samples to use this component over a remote pull from bedrock itself.  Also isolates common infrastructure provisioning into a separate environment called `azure-common-infra` such that it can be independently provisioned outside of the cluster creation process.

### CODE REVIEW

File: cluster/environments/azure-simple/main.tf [hunk 1]
```diff
@@ -1,61 +1,29 @@
-provider "azurerm" {
-  subscription_id = "${var.subscription_id}"
-  client_id       = "${var.service_principal_id}"
-  client_secret   = "${var.service_principal_secret}"
-  tenant_id       = "${var.tenant_id}"
+terraform {
```
Comment (dtzar) [inline cluster/environments/azure-simple/main.tf]: We shouldn't require a backend for `azure-simple` IMO.  It requires extra setup / time and more problems for our CI system.  It isn't very hard to setup a backend and some terraform code to provision it + docs is enough IMO.  If we do want to keep it somewhere (i.e. `azure-advanced`) then we should provision the storage account for the backend as an option in our repo.

File: cluster/environments/azure-simple/main.tf [hunk 2]
```diff
@@ -1,61 +1,29 @@
...
-  }
+resource "azurerm_resource_group" "cluster_rg" {
+  name     = "${var.resource_group_name}"
+  location = "${var.resource_group_location}"
 }
 
-module "aks" {
-  source = "../../azure/aks"
+module "aks-gitops" {
+  source = "github.com/timfpark/bedrock//cluster/azure/aks-gitops?ref=aks_gitops"
```
Comment (dtzar) [inline cluster/environments/azure-simple/main.tf]: Needs link for url when merged to master

File: cluster/environments/azure-simple/main.tf [hunk 3]
```diff
@@ -1,61 +1,29 @@
...
-  gitops_path            = "${var.gitops_path}"
-}
-
-module "kubediff" {
-    source = "../../common/kubediff"
-
-    kubeconfig_complete       = "${module.aks.kubeconfig_done}"
-    gitops_ssh_url            = "${var.gitops_ssh_url}"
+  ssh_public_key           = "${var.ssh_public_key}"
+  vnet_subnet_id           = "${var.vnet_subnet_id}"
```
Comment (dtzar) [inline cluster/environments/azure-simple/main.tf]: This should be gathered from the provisioning, not a var

File: cluster/environments/azure-simple/main.tf (hunk 1 above)
Comment (timfpark) [reply cluster/environments/azure-simple/main.tf]: Agreed - I will remove this.

File: cluster/environments/azure-simple/main.tf (hunk 2 above)
Comment (timfpark) [reply cluster/environments/azure-simple/main.tf]: Yes - agreed - will fix this on merge to Microsoft/bedrock

File: cluster/environments/azure-simple/main.tf (hunk 3 above)
Comment (timfpark) [reply cluster/environments/azure-simple/main.tf]: Yes, I added a vnet provisioner and obtain the subnet id from it.

**Comment by dtzar** [comment]
Need to fix the tests to pass for `azure-simple`
```
TestIT_Bedrock_AzureSimpretry.go:69: Running terraform [init -upgrade=false]
TestIT_Bedrock_AzureSimpcommand.go:53: Running command terraform with args [init -upgrade=false]
TestIT_Bedrock_AzureSimpcommand.go:121: ‌Initializing modules...‌
TestIT_Bedrock_AzureSimpcommand.go:121: - module.vnet
TestIT_Bedrock_AzureSimpcommand.go:121:   Getting source "github.com/Microsoft/bedrock/cluster/azure/vnet"
TestIT_Bedrock_AzureSimpcommand.go:121: - module.aks-gitops
TestIT_Bedrock_AzureSimpcommand.go:121:   Getting source "github.com/Microsoft/bedrock/cluster/azure/aks-gitops"
TestIT_Bedrock_AzureSimpcommand.go:121: ‌Error downloading modules: Error loading modules: subdir &quot;cluster&#x2F;azure&#x2F;aks-gitops&quot; not found‌
TestIT_Bedrock_AzureSimpretry.go:77: Returning due to fatal error: FatalError{Underlying: exit status 1}
TestIT_Bedrock_AzureSimpretry.go:69: Running terraform [destroy -auto-approve -input=false -lock=false -var ssh_public_key="ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC2O3KDZTxoWFPVJHQziv8p3Qh/MfzHZ1gnYK9Nq09JjOxMsLhE7yPhPAnL0qwOMog+o1/v9dc7Aq2Xy6RjEpXoBi/5O4gCRaEpzr0GtvR2p/f+aZ+BfShxpE14f0MglenIRov5fgj+PhNHmNBf6k6CumlbiEh4Qlz9zAdOyRGFf3EZfUa2PeC4SjZyVBTNaaAuIjJ2muQP8KKK5eJ9wwZlHjVP/nP7di15fxoGLXrv0Auw0Ckp5ZPzLmeI5kXyDuLI3CIZu/cYXp0JZa86V/1n7h/y/uYhtrryMNsGr0evQYDf7Wu0rBHlLX8DGiznGWIq8uAJdEG+dP2XptJh/i+f vsts@fv-az619" -var resource_group_name="gTestk8s-IW7LNI-rg" -var resource_group_location="eastus2" -var dns_prefix="gTestk8s-IW7LNI-dns" -var service_principal_id="***" -var service_principal_secret="***" -var cluster_name="gTestk8s-IW7LNI" -var gitops_ssh_url="git@github.com:timfpark/fabrikate-cloud-native-materialized.git" -var gitops_ssh_key="/home/vsts/work/1/s/gopath/src/github.com/Microsoft/bedrock/test/id_rsa.pub" -var tenant_id="***" -var subscription_id="***"]
TestIT_Bedrock_AzureSimpcommand.go:53: Running command terraform with args [destroy -auto-approve -input=false -lock=false -var ssh_public_key="ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC2O3KDZTxoWFPVJHQziv8p3Qh/MfzHZ1gnYK9Nq09JjOxMsLhE7yPhPAnL0qwOMog+o1/v9dc7Aq2Xy6RjEpXoBi/5O4gCRaEpzr0GtvR2p/f+aZ+BfShxpE14f0MglenIRov5fgj+PhNHmNBf6k6CumlbiEh4Qlz9zAdOyRGFf3EZfUa2PeC4SjZyVBTNaaAuIjJ2muQP8KKK5eJ9wwZlHjVP/nP7di15fxoGLXrv0Auw0Ckp5ZPzLmeI5kXyDuLI3CIZu/cYXp0JZa86V/1n7h/y/uYhtrryMNsGr0evQYDf7Wu0rBHlLX8DGiznGWIq8uAJdEG+dP2XptJh/i+f vsts@fv-az619" -var resource_group_name="gTestk8s-IW7LNI-rg" -var resource_group_location="eastus2" -var dns_prefix="gTestk8s-IW7LNI-dns" -var service_principal_id="***" -var service_principal_secret="***" -var cluster_name="gTestk8s-IW7LNI" -var gitops_ssh_url="git@github.com:timfpark/fabrikate-cloud-native-materialized.git" -var gitops_ssh_key="/home/vsts/work/1/s/gopath/src/github.com/Microsoft/bedrock/test/id_rsa.pub" -var tenant_id="***" -var subscription_id="***"]
TestIT_Bedrock_AzureSimpcommand.go:121: ‌
TestIT_Bedrock_AzureSimpcommand.go:121: ‌Error: ‌Error loading modules: subdir &quot;cluster&#x2F;azure&#x2F;aks-gitops&quot; not found‌
TestIT_Bedrock_AzureSimpcommand.go:121: 
TestIT_Bedrock_AzureSimpcommand.go:121: ‌
TestIT_Bedrock_AzureSimpretry.go:77: Returning due to fatal error: FatalError{Underlying: exit status 1}
--- FAIL: TestIT_Bedrock_AzureSimple_Test (5.36s)
    apply.go:13: FatalError{Underlying: exit status 1}
    destroy.go:11: FatalError{Underlying: exit status 1}
FAIL
exit status 1
FAIL	github.com/microsoft/bedrock	5.368s
```

**Comment by timfpark** [comment]
Test failures are due to a chicken and egg problem with the remote module path.

**Pull request merged**